### PR TITLE
Checkout email templates repo to disk.

### PR DESCRIPTION
**Pull request by gnott**
Re issue https://github.com/elifesciences/issues/issues/6595

This is my attempt at loading a repo onto the instance, cribbed from the `strip-coverletter` example. There's probably more than one way to do this, over to you please @lsh-0. Will the repo being marked as `private` causing any additional problems?

### CODE REVIEW

**Comment by lsh-0** [comment]
> Will the repo being marked as private causing any additional problems?

probably, but we'll cross that bridge when we get to it.

File: salt/elife-bot/email-templates.sls [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# email templates
+#
+
+install-email-templates:
+    git.latest:
```
Comment (lsh-0) [inline salt/elife-bot/email-templates.sls:6]: you might want to crib from here if you have problems:
https://github.com/elifesciences/elife-bot-formula/blob/master/salt/elife-bot/init.sls#L13-L21

I'd also recommend adding the force_* params to keep that repo free from accidental modifications.

File: salt/example.top [hunk 2]
```diff
@@ -4,6 +4,7 @@ base:
         - elife.redis-server
         - elife.docker
         - elife-bot.strip-coverletter
+        - elife-bot.email-templates
```
Comment (lsh-0) [inline salt/example.top:7]: it doesn't have to live in a separate state file. If elife-bot needs it then it can live just as happily in `init.sls`

**Review by lsh-0 — APPROVED** [review]

File: salt/example.top (hunk 2 above)
Comment (lsh-0) [reply salt/example.top:7]: you'll need to update the top file in builder-configuration as well.

Comment (lsh-0) [reply salt/example.top:7]: https://github.com/elifesciences/builder-configuration/blob/master/salt/top.sls#L63-L92

Comment (gnott) [reply salt/example.top:7]: I could avoid changing that if I just add this to the `init.sls`?

**Comment by gnott** [comment]
I think the test pipeline has stalled at trying to checkout the private repo...

**Comment by gnott** [comment]
I'll check which users have access to the repo.

**Comment by gnott** [comment]
Er, I don't see settings on https://github.com/elifesciences/elife-email-templates - could you please check when you are able @lsh-0 whether Alfred etc. user has acccess to the repo?

**Comment by lsh-0** [comment]
I've made you admin for that repo

**Comment by lsh-0** [comment]
I've added 'butlers' with read access. Not sure if it needs more or less access (or a different user entirely).

**Comment by gnott** [comment]
Thanks for the update Luke. I'm aborting the PR pipeline run and will run it again to see if that might do it.

File: salt/example.top (hunk 2 above)
Comment (lsh-0) [reply salt/example.top:7]: yes

Comment (lsh-0) [reply salt/example.top:7]: we typically split them into logical units when different environments need different things, but if all environments need it then just add it to init.sls

**Comment by lsh-0** [comment]
looks like it died in the same place twice now:

```
[2021-05-13T00:08:01.558Z] [34.204.13.7] out: [INFO    ] Running state [./strip-coverletter-docker.sh /opt/strip-coverletter/dummy.pdf /tmp/dummy.pdf && rm /tmp/dummy.pdf] at time 00:07:57.093166
[2021-05-13T00:08:01.558Z] [34.204.13.7] out: [INFO    ] Executing state cmd.run for [./strip-coverletter-docker.sh /opt/strip-coverletter/dummy.pdf /tmp/dummy.pdf && rm /tmp/dummy.pdf]
[2021-05-13T00:08:01.558Z] [34.204.13.7] out: [INFO    ] Executing command './strip-coverletter-docker.sh /opt/strip-coverletter/dummy.pdf /tmp/dummy.pdf && rm /tmp/dummy.pdf' as user 'elife' in directory '/opt/strip-coverletter'
[2021-05-13T00:26:31.903Z] [34.204.13.7] out: [INFO    ] {'pid': 12967, 'Sending interrupt signal to process
[2021-05-13T00:26:37.827Z] /srv/builder/bldr: line 22: 28077 Terminated              python src/taskrunner.py "$*"
```

I know the strip-coverletter step isn't pulling from dockerhub but is instead building it the image so takes longer.

**Comment by gnott** [comment]
I think because the git repo was specified as `https://...` it may not have been authenticating with the key. I pushed a change to see if it might get further.

In the end, I'll probably create a fresh branch and new PR once the right combination is proving green.

**Comment by gnott** [comment]
An observation, this (and the bug fix PR https://github.com/elifesciences/elife-bot-formula/pull/71) correctly resulted in the files ending up in `/opt/elife-email-templates`, but the owner is `root:root`. They are world readable, though, so I think it should be fine they are not writable by the `elife` user, it might be better for them to be read-only by the user that runs worker processes.

**Pull request merged**